src/modeler/AutoRest.Swagger/Validation/AvoidNestedProperties.cs

```diff
@@ -52,5 +52,17 @@ private static bool IsClientFlattenUsed(Dictionary<string, object> extensions)
         ///     The severity of this message (ie, debug/info/warning/error/fatal, etc)
         /// </summary>
         public override Category Severity => Category.Warning;
+
+        /// <summary>
+        /// What kind of open api document type this rule should be applied to
+        /// </summary>
+        public override ServiceDefinitionDocumentType ServiceDefinitionDocumentType => ServiceDefinitionDocumentType.ARM;
```

Comment:
In this case, I'd be incline to add data plane too. For other rules, sure, we can discus and figure out.